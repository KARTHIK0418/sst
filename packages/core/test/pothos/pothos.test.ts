import { execSync } from 'child_process';
import { expect, test } from 'vitest';

test('extracts schema', async () => {
  const res = execSync('node ./pothos.mjs', {
    cwd: __dirname,
    encoding: 'utf8',
  });
  expect(res).toMatchSnapshot();
});
