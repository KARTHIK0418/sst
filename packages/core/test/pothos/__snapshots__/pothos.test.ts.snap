// Vitest Snapshot v1

exports[`extracts schema 1`] = `
"// schema.ts
import SchemaBuilder from \\"@pothos/core\\";
var builder = new SchemaBuilder({});
const _DUMMY_RESOLVER = {
  serialize: x => x,
  parseValue: x => x
};
builder.scalarType(\\"AddScalarType\\", _DUMMY_RESOLVER);
builder.scalarType(\\"ScalarType\\", _DUMMY_RESOLVER);
builder.objectType(class AlphaClass {}, {
  name: \\"AlphaClass\\",
  fields: t => ({
    test: t.exposeID(\\"id\\")
  })
});
builder.objectType(class BetaClass {}, {
  name: \\"BetaClass\\",
  fields: t => ({
    test: t.exposeID(\\"id\\")
  })
});

var MyEnum = /* @__PURE__ */(MyEnum2 => {
  MyEnum2[MyEnum2[\\"FOO\\"] = 0] = \\"FOO\\";
  MyEnum2[MyEnum2[\\"BAR\\"] = 1] = \\"BAR\\";
  return MyEnum2;
})(MyEnum || {});

builder.enumType(MyEnum, {
  name: \\"MyEnum\\"
});
builder.queryType({});
builder.mutationType({});
var schema = builder.toSchema({});
"
`;
